### PR TITLE
SW-4695 Allow click/popup interactions on zone cutting in planting site editor

### PR DESCRIPTION
**Pull request by karthikbtf**
- added popup mechanism with selection
- auto pops up on left most new zone upon cutting
- interactive clicking with popup
- added a popup component that looks like a modal with header and footer
- follow up PRs will populate this component to allow setting name and planting density of selected zones
- see attached video

screencast 2024-01-11 19-07-42

### CODE REVIEW

**Comment by karthikbtf** [comment]
@nickgraz when you get a chance, I have a bunch of PRs open #2028 to #2032 .. have more coming up for later.. just a heads up.

File: src/components/Map/EditableMapV2.tsx [hunk 1]
```diff
@@ -41,24 +43,23 @@ export default function EditableMap({
...
   const theme = useTheme();
   const [mapViewStyle, onChangeMapViewStyle] = useMapViewStyle();
-
-  useEffect(() => {
-    // `firstVisible` detects when the box containing the map is first visible in the viewport. The map should only be
-    // rendered if `firstVisible` is true. This accounts for cases in which the map is initially rendered hidden, and
-    // is improperly resized when it first becomes visible.
-    setFirstVisible((fv) => fv || visible);
-  }, [visible]);
+  let destroying = false;
```
Comment (nickgraz) [inline src/components/Map/EditableMapV2.tsx:62]: Should this be hooked up to state?

Comment (karthikbtf) [reply src/components/Map/EditableMapV2.tsx:62]: I lifted this from [here](https://github.com/terraware/terraware-web/blob/main/src/components/Map/Map.tsx#L439), upon debug/experimentation the map destroy is called after the component is unmounted (the state is no longer accessible). This is hacky, I can remove the whole destroying logic and see if anything barfs when I clean things up?

File: src/components/PlantingSites/editor/Zones.tsx [hunk 2]
```diff
@@ -94,6 +153,30 @@ export default function Zones(props: ZonesProps): JSX.Element {
...
+
+        const onUpdate = (nameVal: string, targetPlantingDensityVal: number) => {
+          // TODO
+          return;
+        };
+
+        return (
+          <TooltipContents
+            name={name}
+            onClose={() => onClose?.()}
```
Comment (nickgraz) [inline src/components/PlantingSites/editor/Zones.tsx:170]: 👀 I didn't know you could do that, nice

Comment (karthikbtf) [reply src/components/PlantingSites/editor/Zones.tsx:170]: wait, i thought i saw this in your PR and lifted the idea from there.. lol

File: src/components/PlantingSites/editor/Zones.tsx [hunk 3]
```diff
@@ -77,9 +106,39 @@ export default function Zones(props: ZonesProps): JSX.Element {
...
+        if (cutZones) {
+          const idGenerator = IdGenerator(cutZones);
+          const zonesWithIds = cutZones.map((zone) => toZoneFeature(zone, idGenerator)) as GeometryFeature[];
+
+          setZones({
+            type: 'FeatureCollection',
+            features: zonesWithIds,
+          });
+
+          const leftMostNewZone = zonesWithIds
```
Comment (nickgraz) [inline src/components/PlantingSites/editor/Zones.tsx:120]: This seems like a good candidate for its own util function and a test

File: src/components/PlantingSites/editor/Zones.tsx (hunk 2 above)
Comment (nickgraz) [reply src/components/PlantingSites/editor/Zones.tsx:170]: Haha oh I wish. I usually do `onClose && onClose()`

File: src/components/PlantingSites/editor/Zones.tsx (hunk 3 above)
Comment (karthikbtf) [reply src/components/PlantingSites/editor/Zones.tsx:120]: Good call, will update this PR with util/test and also the cutPolygons test.

File: src/components/Map/EditableMapV2.tsx (hunk 1 above)
Comment (karthikbtf) [reply src/components/Map/EditableMapV2.tsx]: Removed this destroy logic for now, not sure it's needed.

**Review by nickgraz — APPROVED** [review]

**Pull request merged**